File: academic-system-python/academic_system/controller/authentication_controller.py
from ..model.user import User
from ..security.authentication_service import AuthenticationService


class AuthenticationController:
    def __init__(self, authentication_service: AuthenticationService):
        self._authentication_service = authentication_service

    def login(self, username: str, password: str) -> User:
        return self._authentication_service.authenticate(username, password)

    def logout(self) -> None:
        self._authentication_service.logout()

File: academic-system-java/src/main/java/org/example/academic/system/controller/AuthenticationController.java
package org.example.academic.system.controller;

import org.example.academic.system.model.User;
import org.example.academic.system.security.AuthenticationService;

public class AuthenticationController {
    private final AuthenticationService authenticationService;

    // Injeta o serviço de autenticação
    public AuthenticationController(AuthenticationService authenticationService) {
        this.authenticationService = authenticationService;
    }

    // Autentica o usuário com as credenciais fornecidas e retorna o usuário logado
    public User login(String username, String password) {
        return authenticationService.authenticate(username, password);
    }

    // Encerra a sessão do usuário autenticado
    public void logout() {
        authenticationService.logout();
    }
}
